opencensus/metrics/export/summary.py

```diff
@@ -0,0 +1,143 @@
+
+    :type count: long
+    :param count: the count of the population values.
+
+    :type sum_data: float
+    :param sum_data: the sum of the population values.
+
+    :type snapshot: Snapshot
+    :param snapshot: the bucket boundaries of a histogram.
+    """
+
+    def __init__(self, count, sum_data, snapshot):
+        check_count_and_sum(count, sum_data)
+        self._count = count
+        self._sum_data = sum_data
+
+        if snapshot is None:
+            raise ValueError('snapshot must not be none')
+
+        self._snapshot = snapshot
+
+    @property
+    def count(self):
+        """Returns the count of the population values"""
+        return self._count
+
+    @property
+    def sum_data(self):
+        """Returns the sum of the population values."""
+        return self._sum_data
+
+    @property
+    def snapshot(self):
+        """Returns the bucket boundaries of a histogram."""
+        return self._snapshot
+
+
+class Snapshot(object):
+    """Represents the summary observation of the recorded events over a
+    sliding time window.
+
+    :type count: long
+    :param count: the number of values in the snapshot.
+
+    :type sum_data: float
+    :param sum_data: the sum of values in the snapshot.
+
+    :type value_at_percentiles: ValueAtPercentile
+    :param value_at_percentiles: a list of values at different percentiles
+    of the distribution calculated from the current snapshot. The percentiles
+    must be strictly increasing.
+    """
+
+    def __init__(self, count, sum_data, value_at_percentiles=None):
+        check_count_and_sum(count, sum_data)
+        self._count = count
+        self._sum_data = sum_data
+
+        if value_at_percentiles is None:
+            value_at_percentiles = []
+
+        if not isinstance(value_at_percentiles, list):
+            raise ValueError('value_at_percentiles must be an '
+                             'instance of list')
+
+        self._value_at_percentiles = value_at_percentiles
+
+    @property
+    def count(self):
+        """Returns the number of values in the snapshot"""
+        return self._count
+
+    @property
+    def sum_data(self):
+        """Returns the sum of values in the snapshot."""
+        return self._sum_data
+
+    @property
+    def value_at_percentiles(self):
+        """Returns a list of values at different percentiles of the distribution
+        calculated from the current snapshot.
+        """
+        return self._value_at_percentiles
+
+
+class ValueAtPercentile(object):
+    """Represents the value at a given percentile of a distribution.
+
+    :type percentile: float
+    :param percentile: the percentile in the ValueAtPercentile.
+
+    :type value: float
+    :param value: the value in the ValueAtPercentile.
+    """
+
+    def __init__(self, percentile, value):
+
+        if not 0 < percentile <= 100.0:
+            raise ValueError("percentile must be in the interval (0.0, 100.0)")
```

Comment:
Might want this to be `(0.0, 100.0]` to match the check (and the [spec](https://github.com/census-instrumentation/opencensus-proto/blob/1aa555954f538f658b483ab6190e393f4aacf733/src/opencensus/proto/metrics/v1/metrics.proto#L288)).